src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -150,9 +150,18 @@ private Task InitializeWebHostRuntimeChannelsAsync()
             return Task.CompletedTask;
         }
 
-        private bool ShouldStartInPlaceholderMode()
+        internal bool ShouldStartInPlaceholderMode()
         {
-            return string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled();
+            if (string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled())
+            {
+                if (_environment.IsAppServiceWindowsEnvironment())
+                {
+                    string siteName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteName);
+                    return siteName.Equals(ScriptConstants.PlaceholderTemplateSiteName, StringComparison.InvariantCultureIgnoreCase);
+                }
+                return true;
```

Comment:
Nit, but do you mind adding comments here? Like this "true" is for Linux that is in placeholder mode with no worker runtime set, for example?